### ISSUE TITLE
Exception in addNewTargetLines

### ISSUE THREAD

**Issue by videophonegeek** [jbreckmckye/raphael-paragraph#1]
Just trying this out for the first time. I get an exception on the this.addLine();

```
        function addNewTargetLines() {
            var extraLinesToAdd = targetLines.length - currentLines.length;
            var newLineText;
            if (extraLinesToAdd >= 1) {
                for (var i = 0; i < extraLinesToAdd; i++) {
                    newLineText = targetLines[currentLines.length];
                    this.addLine();
                    this.addTextToLine(newLineText);
                }
            }
        }
```

Trace:
[Error] TypeError: undefined is not a function (evaluating 'this.addLine()')
    addNewTargetLines (raphael-paragraph.js, line 395)
    restoreState (raphael-paragraph.js, line 371)
    restoreState (raphael-paragraph.js, line 140)
    addSpaceAndTruncatedWord (raphael-paragraph.js, line 583)
    rollbackWordsUntilEllipsisFits (raphael-paragraph.js, line 622)
    ellipsizePreviousWord (raphael-paragraph.js, line 602)
    (anonymous function) (raphael-paragraph.js, line 223)
    arrayForEach (raphael-paragraph.js, line 307)
    tryEachStrategyUntilOneMakesWordFitSpace (raphael-paragraph.js, line 221)
    tryWordStrategies (raphael-paragraph.js, line 212)
    addWordUsingStrategies (raphael-paragraph.js, line 202)
    addWordsUntilOutOfSpace (raphael-paragraph.js, line 182)
    arrayForEach (raphael-paragraph.js, line 307)
    fitWordsIntoSpace (raphael-paragraph.js, line 180)
    paragraph (raphael-paragraph.js, line 75)
    (anonymous function) (test.html, line 36)
    fire (jquery-2.1.4.js, line 3099)
    fireWith (jquery-2.1.4.js, line 3211)
    ready (jquery-2.1.4.js, line 3417)
    completed (jquery-2.1.4.js, line 3433)

Here is the actual text which I kept editing until the exception went away:
"SUPPOSE YOU COUGH, GO DOCTOR YOU?" --exception
"SUPPOSE YOU COUGH" -exception
"SUPPOSE YOU" -works

**Comment by jbreckmckye** [comment]
I've just made a change to something that looks a bit suspicious - can you confirm whether it now works?

If not, can you tell me the config options you're passing? Thanks.

**Comment by videophonegeek** [comment]
I tried the latest with the same results.

Here is the config:
var paper= Raphael("holder", "100%", "96");  
var answer4 = paper.paragraph({text : "SUPPOSE YOU COUGH, GO DOCTOR YOU?"});

**Comment by jbreckmckye** [comment]
Ah, looks like I forgot to build out the source change to the distributable. Maybe try again? Sorry about that.

**Comment by videophonegeek** [comment]
I simply replaced the distributable raphael-paragraph.js with the fresh one and the behavior changed. The wrap displays one word per line, no seeming to mind the maxWidth.

var answer4 = paper.paragraph({x : 100, y : 96, maxWidth : 400, text : "SUPPOSE YOU COUGH, GO DOCTOR YOU?"});

The original exception does not appear but a new one does:
[Error] Raphael.fn.paragraph was already defined when the raphael-paragraph plugin was run.
    (anonymous function) (raphael-paragraph.js, line 841)
    **webpack_require** (raphael-paragraph.js, line 811)
    (anonymous function) (raphael-paragraph.js, line 831)
    (anonymous function) (raphael-paragraph.js, line 832)
    **webpack_require** (raphael-paragraph.js, line 20)
    (anonymous function) (raphael-paragraph.js, line 49)
    **webpack_require** (raphael-paragraph.js, line 20)
    (anonymous function) (raphael-paragraph.js, line 40)
    global code (raphael-paragraph.js, line 41)

**Comment by jbreckmckye** [comment]
I'm at work right now, so I can't look at this immediately, but I will give this a thorough investigation tonight.

**Comment by videophonegeek** [comment]
I'm in Japan at work so look for your response in the morning. After we get past this one I have some other questions.

```
  From: Jimmy Breck-McKye <notifications@github.com>
```

 To: jbreckmckye/raphael-paragraph raphael-paragraph@noreply.github.com 
Cc: John Feagans feagans_john@yahoo.com 
 Sent: Tuesday, August 25, 2015 6:27 PM
 Subject: Re: [raphael-paragraph] Exception in addNewTargetLines (#1)

I'm at work right now, so I can't look at this immediately, but I will give this a thorough investigation tonight.—

**Comment by jbreckmckye** [comment]
Hello again, John. Thanks for being patient.

So, there were actually two issues here:

**The last distributable was not 'compiled' correctly by Webpack**
It would seem that webpack 'compiled' the last build improperly; it's not clear why, but a module was calling itself as a dependency. This was obviously no good but fairly easy to fix.

**The percentage width set on the paper constructor threw the plugin off**
By this, I mean that setting a width of '100%' on the Raphael constructor created problems later when Raphael-paragraph tried to default its max width and couldn't properly sniff the dimensions of the container.

Technically, percentage string widths are not listed as a valid argument for `Raphael()` in [the documentation](http://raphaeljs.com/reference.html#Raphael), but it looks like there's a lot of example code doing it in the wild. So it was prudent to make Raphael-paragraph use a more robust way to check the width of its container.

I've now fixed this and the changes should be in the new distributable.

> After we get past this one I have some other questions.

If your code now works - which it should, testing on Chrome and IE - then fire away. I don't think many people have used this library so far, so I'm eager to hear how someone might use it.

**Comment by videophonegeek** [comment]
Solved!

I have two new issues to open (later)

I am using this in a responsive design that works on all platforms including Android and iOS as native apps using the Intel XDK. I tested today with Safari on OSX, Android on Nexus 7, and iOS on iPhone 6.  Lack of a text wrap made this an impossible screen geometry exercise on mobile screens. I use Raphael animations to bring multiple choice tests in and animate the correct/incorrect choices.
